src/BaseExportAdapter.php

```diff
@@ -51,8 +56,7 @@ public function export(ExportConfig $exportConfig, string $csvFilePath): ExportR
                 $query,
                 $exportConfig->getMaxRetries(),
                 function (QueryResult $result) use ($exportConfig, $csvFilePath) {
-                    $resultWriter = new QueryResultCsvWriter($this->state);
-                    return $resultWriter->writeToCsv($result, $exportConfig, $csvFilePath);
+                    return $this->resultWriter->writeToCsv($result, $exportConfig, $csvFilePath);
```

Comment:
Tu sa povodne `ResultWriter` (`QueryResultCsvWriter`) vytvaral, teraz je to uz atribut triedy.